compiler-core/src/build/project_compiler.rs

```diff
@@ -68,6 +70,10 @@ where
             .into_iter()
             .map(|p| (p.name.to_string(), p))
             .collect();
+
+        // TODO can this return a result?
+        let lock = NamedLock::create("gleam-compile")?;
```

Comment:
This cannot be a result, is there a better way todo it? Does this name mean anything?

Reply:
Perhaps replace `?` with `.expect("Could not lock build directory")`